tiledb/sm/query/query_condition.cc

```diff
@@ -1061,165 +1083,186 @@ void QueryCondition::apply_ast_node_dense(
   const bool nullable = attribute->nullable();
 
   // Process the validity buffer now.
-  if (nullable) {
+  if (nullable && node->get_condition_value_view().content() == nullptr) {
```

Comment:
I remember we added this because it would remove the nullable conditional from the later cell by cell processing and improved performance for one of our customers scenarios. With ANDs, it worked fine but with OR, I see how this doesn't work. But, instead of plumbing the nullable checks down to the cell by cell layer, can we do a post processing similar to this after the full OR tree is processed?